File: src/app/service/employee-response.ts
import {Position} from "./position-enum";

export interface EmployeeResponse {
  id: number;
  fullName: string;
  email: string;
  dateOfBirth: string;
  startDate: string;
  title: string;
  position: Position;
  salary: number;
  jopName?: string | null;
  clientName?: string | null;
}

File: backend/src/main/java/com/application/cv_application/response/EmployeeResponse.java
package com.application.cv_application.response;

import com.application.cv_application.enums.Position;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class EmployeeResponse {
    private Integer id;
    private String fullName;
    private String email;
    private LocalDate dateOfBirth;
    private LocalDate startDate;
    private String title;
    private Position position;
    private Integer salary;
    private String jopName;
    private String clientName;
    private Boolean isActive;
}

